core/src/main/java/org/jruby/RubyThread.java

```diff
@@ -940,9 +940,18 @@ public IRubyObject getName() {
         return RubyString.newString(runtime, rubyName);
     }
 
+    /**
```

Comment:
I think I would be interested in seeing this fall into org.jruby.api somewhere.  I have been trying to get all methods people would use in native extensions into this package space.  I have also been dogfooding our internals to use these same APIs when it makes sense.

org.jruby.api.Runtime#currentContext(Ruby) or something like that?  Not sure on either the class name or the method name but just putting that out there.

The other broad point is to kill off any external consumers from hitting Ruby methods at all.   I would love to never see us use Ruby as a reference anywhere some day but I think needing to get current ThreadContext will be with us for a long time.

Reply:
Just to add here that 9.4 has none of the new .api packages but the plan is to backport those files so we have a migration path for 9.4 and possibly 9.3 for native extensions.

Reply:
> I would love to never see us use Ruby as a reference anywhere some day

it's a multi-tenant runtime so I am not sure how that would end up being achieved...

what I can tell, as I've been on multiple bigger projects consuming and mixing Ruby and Java is that the runtime is usually something that is kept externally or queried frequently `Ruby.getGlobalRuntime` (even if it's a single runtime use-case).

> seeing this fall into `org.jruby.api` somewhere

would be nice, but it's a big chunk IMHO, the only way to cleanup the distinction would be to have interfaces on objects that Ruby users from Java encounter e.g. `RubyArray implements IRubyArray`, `RubyHash`, `RubyThread` etc. having just the one `IRubyThread` added here, with a static method, doesn't get us far...

I can remove the extra `current(final Ruby runtime)` in this PR and can continue consuming `current(IRubyObject recv)`, it's mostly for convenience

Reply:
@kares yeah for embedding with multiple runtimes you have to use runtime.  I meant for internal use and java native extension gems.  Even then I know we will still need something to get TC from runtime.

Also I am only talking about this single method `current` so we don't have another way of getting TC.  If you just move this to a new class with a static method:  org.jruby.api.VM.currentThread(Ruby runtime)  then I think it is fine.

Reply:
I've reverted the change for now (the work-around with `RubyThread.curren(IRubyObject)` will do, it's likely a minor use-case anyways).
I think I get what you mean but I believe it deserves a proper API design, just moving a static method somewhere while still returning `RubyThread` sounds a bit half-baked and I can not commit to the full thingy atm.